api/lib/infrastructure/repositories/user-repository.js

```diff
@@ -125,6 +170,7 @@ function _toDomain(userBookshelf) {
     certificationCenterMemberships: _toCertificationCenterMembershipsDomain(userBookshelf.related('certificationCenterMemberships')),
     pixRoles: _toPixRolesDomain(userBookshelf.related('pixRoles')),
     hasSeenAssessmentInstructions: Boolean(userBookshelf.get('hasSeenAssessmentInstructions')),
+    authenticationMethods: _toAuthenticationMethodsDomain(userBookshelf.related('authenticationMethods')),
```

Comment:
**Suggestion**: J'ai continué à réfléchir sur l'ajout de l'attribut `authenticationMethods` ici et l'impact inévitable sur le serializer et donc le/les fronts qui utilisent ces données. Je pense qu'il aurait été plus simple de conserver les attributs `password` et `shoudlChangePassword` sur le model `User` et de les renseigner depuis la table authentication-methods. Ca réduit, je pense, les changements apportés au code en limitant les changements au répo voire aux usecases. Si tu penses que ce n'est pas la bonne solution, il faut au moins suppimer password et shouldChangePassword du mofel User afin d'éviter les endroits où on pourrait encore lire dans ces colonnes par erreur vu qu'on ne les supprime pas encore.

Reply:
Dans le domain model `User`
* `password` et `shouldChangePassword` ont été supprimés
* `authenticationMethods` a été ajouté

Après plusieurs analyses, j'ai voulu garder la logique prise pour `GAR` et `Pole emploi` : déplacer complètement les attributs et type de connexion vers un model distinct `AuthenticationMethod`.
J'ai fait le choix de modifier au minimum les `usecases`, mais de travailler plus sur les interfaces/fonctions des `repositories` et `services` concernés.

Reply:
Voir plus haut.